sdk/core/core-http/review/core-http.api.md

```diff
@@ -0,0 +1,980 @@
+export abstract class BaseRequestPolicy implements RequestPolicy {
+    protected constructor(_nextPolicy: RequestPolicy, _options: RequestPolicyOptionsLike);
+    log(logLevel: HttpPipelineLogLevel, message: string): void;
+    // (undocumented)
+    readonly _nextPolicy: RequestPolicy;
+    // (undocumented)
+    readonly _options: RequestPolicyOptionsLike;
+    // (undocumented)
+    abstract sendRequest(webResource: WebResourceLike): Promise<HttpOperationResponse>;
+    shouldLog(logLevel: HttpPipelineLogLevel): boolean;
+}
+
+// @public (undocumented)
+export class BasicAuthenticationCredentials implements ServiceClientCredentials {
+    constructor(userName: string, password: string, authorizationScheme?: string);
+    // (undocumented)
+    authorizationScheme: string;
+    // (undocumented)
+    password: string;
+    signRequest(webResource: WebResourceLike): Promise<WebResourceLike>;
+    // (undocumented)
+    userName: string;
+}
+
+// @public
+export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
+    constructor(nextPolicy: RequestPolicy, options: RequestPolicyOptions, credential: TokenCredential, scopes: string | string[], tokenCache: AccessTokenCache);
+    sendRequest(webResource: WebResourceLike): Promise<HttpOperationResponse>;
+    }
+
+// @public
+export function bearerTokenAuthenticationPolicy(credential: TokenCredential, scopes: string | string[]): RequestPolicyFactory;
+
+// @public (undocumented)
+export interface CompositeMapper extends BaseMapper {
+    // (undocumented)
+    type: CompositeMapperType;
+}
+
+// @public (undocumented)
+export interface CompositeMapperType {
+    // (undocumented)
+    additionalProperties?: Mapper;
+    // (undocumented)
+    className?: string;
+    // (undocumented)
+    modelProperties?: {
+        [propertyName: string]: Mapper;
+    };
+    // (undocumented)
+    name: "Composite";
+    // (undocumented)
+    polymorphicDiscriminator?: PolymorphicDiscriminator;
+    // (undocumented)
+    uberParent?: string;
+}
+
+// @public (undocumented)
+export const Constants: {
+    coreHttpVersion: string;
+    HTTP: string;
+    HTTPS: string;
+    HTTP_PROXY: string;
+    HTTPS_PROXY: string;
+    HttpConstants: {
+        HttpVerbs: {
+            PUT: string;
+            GET: string;
+            DELETE: string;
+            POST: string;
+            MERGE: string;
+            HEAD: string;
+            PATCH: string;
+        };
+        StatusCodes: {
+            TooManyRequests: number;
+        };
+    };
+    HeaderConstants: {
+        AUTHORIZATION: string;
+        AUTHORIZATION_SCHEME: string;
+        RETRY_AFTER: string;
+        USER_AGENT: string;
+    };
+};
+
+// @public (undocumented)
+export function createPipelineFromOptions(pipelineOptions: InternalPipelineOptions, authPolicyFactory?: RequestPolicyFactory): ServiceClientOptions;
+
+// Warning: (ae-forgotten-export) The symbol "FetchHttpClient" needs to be exported by the entry point coreHttp.d.ts
```

Comment:
I fixed most of the ae-forgotten-export warnings.  These two ones I am not sure whether we want to expose `FetchHttpClient`

Reply:
Yeah, let's leave that unexported for now. The work is tracked by #4918